src/ptvsd/common/messaging.py

```diff
@@ -330,7 +331,7 @@ def _send_message(self, type, rest={}):
 
     def send_request(self, command, arguments=None):
         d = {'command': command}
-        if arguments is not None:
+        if arguments is not None and arguments != {}:
```

Comment:
we could use `bool(arguments)` here right?

Reply:
It wouldn't need to be `bool()` then, just `arguments` is the same.

It breaks because it also treats 0 and "" as false. In practice all real messages have dicts as arguments, and it would be fine. But the protocol doesn't require it - it says `any?` - and a bunch of pre-existing tests for messaging code uses integers for  it, including 0, so changing it here requires changing a lot of tests also.